enhancements/proposals/disk_setup.md

```diff
@@ -0,0 +1,225 @@
+# Improve UX around disks
+
+## Summary
+
+Currently, when a user wants to use Scylla with local drives, they need to manually
+create an RAID0 array, format it to XFS and expose it as a PersistentVolume on every Node dedicated to Scylla.
+Formatting operations are already implemented in scripts available in Scylla image.  
+We can use a DaemonSet proposed in [perftune.md](perftune.md) to access and modify physical drives.  
+For exposing PersistentVolumes we may deploy [storage-local-static-provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner) 
+
+Another bad example of UX is io-setup benchmark not being cached. IOSetup takes ~2 minutes to run
+and the result is saved in ephemeral volume, which means it has to be repeated on every
+Pod restart.  
+Scylla recommends, to have the same content of io_properties.yaml (benchmark result) on every node, and
+for known instance types, these values are pre-computed. 
+We may leverage that to speedup even first bootstrap.
+
+## Motivation
+
+Setting it up is not trivial, and different cloud providers have different setup scripts examples.
+Because we recommend to use local drives, we should do our best to have good user experience around it.
+
+### Goals
+
+* Automate creating RAID0
+* Automate setting up XFS partition
+* Automate exposing local disks as PersistentVolumes
+* Cache io-setup benchmark results
+* Use precomputed io-setup benchmark results for known instance types
+
+### Non-Goals
+
+* Setup and format network attached drives.
+
+## Proposal
+
+Three main focus points of this proposals are:
+* Setting up the disks
+* Exposing disks as PersistentVolumes
+* Caching of io-setup results
+
+Each of them have to be optional, because users may have their own solution to each of this problem or our solution may not work 
+on their types of machines.
+
+The first two of those require root access to host machine, third one requires read access to Pod storage.
+These are going to be run by DaemonSet described in [perftune.md](perftune.md), because it 
+has root access to host machine, and it's running on every k8s node having Scylla Pods, so it will
+be able to access both local PVs and network attached PVs. 
+The third one requires access to Pod storage and ability to create ConfigMaps. 
+
+For setting up the disks we are going to execute `scylla_raid_setup` script. 
+This will create a RAID0 array and will format partition to XFS.
```

Comment:
I believe this section needs a description of how the DaemonSet is going to recognize which cloud it's running on and whether we support it. #760 has recently proven that currently it might be impossible to make the disk setup cloud agnostic.